kubernetes/client/kubernetes_test.go

```diff
@@ -0,0 +1,277 @@
+// Copyright © 2018 Genome Research Limited Author: Theo Barber-Bany
+// <tb15@sanger.ac.uk>.
+//
+//  This file is part of wr.
+//
+//  wr is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  wr is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU Lesser General Public License for more details.
+//
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with wr. If not, see <http://www.gnu.org/licenses/>.
+
+package client_test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"strings"
+	"testing"
+	"time"
+
+	apiv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/VertebrateResequencing/wr/kubernetes/client"
+	"github.com/docker/docker/pkg/namesgenerator"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+var tc client.Kubernetesp
+var clientset kubernetes.Interface
+var autherr error
+var testingNamespace string
+
+func init() {
+	tc = client.Kubernetesp{}
+	clientset, _, autherr = tc.Authenticate()
+	if autherr != nil {
+		panic(autherr)
```

Comment:
I'm assuming running this test will always panic if there's no k8s cluster up? Can you please adjust things so that someone can `go test ./...` without k8s and pass? Ie. skip k8s tests (with a skip message) if the necessary stuff isn't in place.